lib/hanami/slice_registrar.rb

```diff
@@ -85,6 +82,16 @@ def with_nested
 
     private
 
+    def ancestors
+      list = []
+      ancestor = parent
+      while ancestor
+        list.unshift(ancestor)
+        ancestor = ancestor.parent
+      end
+      list
+    end
```

Comment:
keen for any input on a better way to do this?

Reply:
I think this is fine as a first cut. I'm sure it could possibly be done via some kind of `reduce` fanciness, but I think this procedural approach is simple to read, and we can always come back and do a targeted refactoring later. My biggest concerns here are about making the _purpose_ of this method super clear, which I think we can address via doc comments.